### PR TITLE
Update GFX908 Decoder ISA-SPEC(02/22/23)

### PR DESCRIPTION
**Pull request by bbiiggppiigg**
Update implementation for AMDGPU GFX908 based on ISA-SPEC 02/22/23

1. Fix use of wrong immediate length in sign_extend (bug-fix)
2. Add memory counter definitions (dsmem)
3. In functions IS_ENC_* use hex representation instead of int
4. Use isArrayIndexValid function to do boundary check for ENC_*_insn_table
5. Fix compile warning for shadowing num_elements in the decoder

### CODE REVIEW

**Review by kupsch — CHANGES_REQUESTED** [review]
See comments in PR #1407.

**Review by kupsch — APPROVED** [review]

File: instructionAPI/src/AMDGPU/gfx908/InstructionDecoder-amdgpu-gfx908.h [hunk 1]
```diff
@@ -73,7 +74,11 @@ namespace Dyninst {
             static const char* bitfieldInsnAliasMap(entryID);
             static const char* condInsnAliasMap(entryID);
 
-
+            //Check if the index (2nd arg) is valid for the array (1st arg)
+            template <typename ArrayType, std::size_t n, typename IndexType>
```
Comment (hainest) [inline instructionAPI/src/AMDGPU/gfx908/InstructionDecoder-amdgpu-gfx908.h:78]: I don't see any usages of this. Should it be in this commit?

Comment (kupsch) [reply instructionAPI/src/AMDGPU/gfx908/InstructionDecoder-amdgpu-gfx908.h:78]: There are many in `instructionAPI/src/AMDGPU/gfx908/amdgpu_gfx908_decoder_impl.C`.

Comment (hainest) [reply instructionAPI/src/AMDGPU/gfx908/InstructionDecoder-amdgpu-gfx908.h:78]: Oh, derp. I didn't notice that file had huge changes and wasn't expanded.

**Review by hainest — APPROVED** [review]

**Pull request merged**